File: articles/the-history-of-data-structures-and-algorithm-analysis/haskell/GrowthRates.hs
module Main where

main :: IO ()
main = mapM_ showGrowth [10.0, 100.0, 1000.0, 10000.0]
  where
    showGrowth n = putStrLn ("n=" ++ show n ++ ", log2=" ++ show (logBase 2 n) ++ ", nlogn=" ++ show (n * logBase 2 n) ++ ", n2=" ++ show (n*n))

File: articles/the-history-of-data-structures-and-algorithm-analysis/rust/growth_rates.rs
fn main() {
    for n in [10.0_f64, 100.0, 1000.0, 10000.0] {
        println!("n={:.0}, log2={:.6}, nlogn={:.6}, n2={:.0}", n, n.log2(), n * n.log2(), n * n);
    }
}
